src/service/alert.service.ts

```diff
@@ -121,7 +121,7 @@ export class AlertService {
     }
 
     closeAlert(id: number, extAlerts?: Alert[]): any {
-        let thisAlerts: Alert[] = extAlerts ? extAlerts : this.alerts;
+        let thisAlerts: Alert[] = (extAlerts && extAlerts.length > 0) ? extAlerts : this.alerts;
```

Comment:
no the fix is still valid. Js truthy check doesnt check for empty its something I overlooked. But I have no idea how this was working all these time as the alerts used to autclose correctly if you remember even in Ng1 with same check